viz/pkg/healthcheck/healthcheck.go

```diff
@@ -40,7 +46,20 @@ type HealthChecker struct {
 func NewHealthChecker(categoryIDs []healthcheck.CategoryID, options *healthcheck.Options) *HealthChecker {
 	parentHC := healthcheck.NewHealthChecker(categoryIDs, options)
 	hc := &HealthChecker{HealthChecker: parentHC}
-	parentHC.AppendCategories(*hc.vizCategory())
+
+	vizCategories := hc.vizCategory()
+	checkMap := map[healthcheck.CategoryID]struct{}{}
+	for _, category := range categoryIDs {
+		checkMap[category] = struct{}{}
+	}
+
+	// Append Only the categories that aare passed
+	for i := range vizCategories {
```

Comment:
I found it a little difficult to read why we needed to go over the list of `vizCategories` to append viz specific checks to the parent healthchecker. I think it might be a little clear if we create a category map from all the viz categories and then loop over `categoryIDs` to figure out what checks we'd need to append. 
```diff
diff --git a/viz/pkg/healthcheck/healthcheck.go b/viz/pkg/healthcheck/healthcheck.go
index 1a476c87..2d1fb9fe 100644
--- a/viz/pkg/healthcheck/healthcheck.go
+++ b/viz/pkg/healthcheck/healthcheck.go
@@ -47,16 +47,15 @@ func NewHealthChecker(categoryIDs []healthcheck.CategoryID, options *healthcheck
 	parentHC := healthcheck.NewHealthChecker(categoryIDs, options)
 	hc := &HealthChecker{HealthChecker: parentHC}

-	vizCategories := hc.vizCategory()
-	checkMap := map[healthcheck.CategoryID]struct{}{}
-	for _, category := range categoryIDs {
-		checkMap[category] = struct{}{}
+	vizCheckMap := map[healthcheck.CategoryID]healthcheck.Category{}
+	for _, category := range hc.vizCategories() {
+		vizCheckMap[category.ID] = category
 	}

-	// Append Only the categories that aare passed
-	for i := range vizCategories {
-		if _, ok := checkMap[vizCategories[i].ID]; ok {
-			parentHC.AppendCategories(vizCategories[i])
+	// only append incoming check categories that are also in the vizCheckMap
+	for _, id := range categoryIDs {
+		if _, ok := vizCheckMap[id]; ok {
+			parentHC.AppendCategories(vizCheckMap[id])
 		}
 	}

@@ -73,7 +72,7 @@ func (hc *HealthChecker) RunChecks(observer healthcheck.CheckObserver) bool {
 	return hc.HealthChecker.RunChecks(observer)
 }

-func (hc *HealthChecker) vizCategory() []healthcheck.Category {
+func (hc *HealthChecker) vizCategories() []healthcheck.Category {
 	return []healthcheck.Category{
 		*healthcheck.NewCategory(LinkerdVizExtensionCheck, []healthcheck.Checker{
 			*healthcheck.NewChecker("linkerd-viz Namespace exists").
```

Reply:
Yep, Makes sense. This was an unnecessary complication after the copy-pasta of the boiler-plate code from the core health-checker.